models/llama/llama/generation.py

```diff
@@ -193,10 +193,17 @@ def generate(
         for cur_pos in range(min_prompt_len, total_len):
             logits = self.model.forward(tokens[:, prev_pos:cur_pos], prev_pos)
             if temperature > 0:
-                probs = self.Math.softmax(logits[:, -1] / temperature, dim=-1)
+                if self.use_triton: 
+                    probs = triton_softmax(logits[:,-1])
+                else: 
+                    probs = self.Math.softmax(logits[:, -1] / temperature, dim=-1)
```

Comment:
there is a mathOps file in this directory to abstract this away from users. Lets use that instead and allow for proper benchmarking (see decorator on the functions there)